src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1542,6 +1542,27 @@ private void visitTypeDeclaration(JSType type, boolean isVarArgs, boolean isOpti
       }
     }
 
+    private boolean emitNoResolvedType(NoType type, String defaultEmit) {
+      if (!opts.partialInput || !type.isNoResolvedType()) {
+        emit(defaultEmit);
+        return true;
+      }
+      // When processing partial inputs, this case handles implicitly forward declared types
+      // for which we just emit the literal type written along with any type parameters.
+      NoResolvedType nType = (NoResolvedType) type;
+      if (nType.getReferenceName() != null) {
+        String displayName = type.getDisplayName();
+        emit(Constants.INTERNAL_NAMESPACE + "." + displayName);
+        List<JSType> templateTypes = nType.getTemplateTypes();
+        if (templateTypes != null && templateTypes.size() > 0) {
+          emitGenericTypeArguments(type.getTemplateTypes().iterator());
+        }
+      } else {
+        emit(defaultEmit);
```

Comment:
can you document when/why this happens, or is that part of the mystery?

Reply:
This code is shared between the type emitter and the extends/implements emitter. It can easily happen for types , when one has type inference involved. We have tests for that. However, for extends/implements I cannot reproduce it with test (i.e. `ClutzBaseClass` doesn't appear anywhere).

Added a todo